### ISSUE TITLE
Page mounted twice on redirect in SPA

### ISSUE THREAD

**Issue by MrWook** [nuxt/nuxt.js#4197]
### Version

[v2.2.0](https://github.com/nuxt.js/releases/tag/v2.2.0)

### Reproduction link

[https://codesandbox.io/s/40y7x1l0p4](https://codesandbox.io/s/40y7x1l0p4)

### Steps to reproduce

1. Open reproduction link
2. Open console
3. Add /random_string to the codesandbox project url like this `https://40y7x1l0p4.sse.codesandbox.io/randome_string`

### What is expected ?

Inside the console the String `test` should only exist once. 

### What is actually happening?

The string `test` will be shown twice.

### Additional comments?

I just added a middleware that redirects to the test page if it is not in the router array:
```
export default ({ app, route, redirect }) => {
  const { path, query } = route;
  const { routes } = app.router.options;
  // Other redirects
  if (routes.find(value => value.path === path) === undefined) {
    return redirect("/test");
  }
};

```
Inside the test page i only output `test` on the mounted event.
This only occure when the nuxt mode is set to `spa`

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8038">#c8038</a>)</em></sub></div>

**Comment by lee-sangyoung** [comment]
I have a same issue

**Comment by aldarund** [comment]
asyncadata/fetch trigger only once in this case.
mounted and created twice. so component got created twice for some reason

**Comment by MrWook** [comment]
Dam Automatic bot. It's still relevant in v2.3.4. Please reopen this.

**Comment by MrWook** [comment]
Because there is no response from the nuxt team i opened a new ticket for it: https://github.com/nuxt/nuxt.js/issues/4757
